content/usgs-packages/sbtools_Discovery.Rmd

```diff
@@ -0,0 +1,224 @@
+
+# verifying where points are supposed to be
+maps::map('usa')
+points(conus$long, conus$lat, col="red", pch=20)
+points(appalachia$long, appalachia$lat, col="green", pch=20)
+
+# query by bounding box
+bbox_sp_obj <- sp::SpatialPoints(appalachia)
+# query_sb_spatial(bbox=bbox_sp_obj)
+
+# query by latitude and longitude vectors
+query_sb_spatial(long = appalachia$long, lat = appalachia$lat)
+query_sb_spatial(long = conus$long, lat = conus$lat)
+
+# query by WKT polygon
+wkt_coord_str <- paste(conus$long, conus$lat, sep=" ", collapse = ",")
+wkt_str <- sprintf("POLYGON((%s))", wkt_coord_str)
+# query_sb_spatial(bb_wkt = wkt_str)
+```
+
+### Using `query_sb_date`
+
+`query_sb_date` returns ScienceBase items that fall within a certain time range. There are multiple timestamps applied to items, so you will need to specify which one to match the range. The default queries are to look for items last updated between 1970-01-01 and today's date. See `?query_sb_date` for more examples of which timestamps are available.
+
+```{r}
+# find data worked on in the last week
+today <- Sys.time()
+oneweekago <- today - (7*24*3600) # days * hrs/day * secs/hr
+recent_data <- query_sb_date(start = today, end = oneweekago)
+head(recent_data, 2)
+
+# find data that's been created over the last year
+oneyearago <- today - (365*24*3600) # days * hrs/day * secs/hr
+recent_data <- query_sb_date(start = today, end = oneyearago, date_type = "dateCreated")
+head(recent_data, 2)
+```
+
+### Using `query_sb_datatype`
+
+`query_sb_datatype` is used to search ScienceBase by the type of data an item is listed as. Run `sb_datatypes()` to get a list of 50 available data types.
+
+```{r}
+# get ScienceBase news items
+sbnews <- query_sb_datatype("News")
+head(sbnews, 2)
+
+# find shapefiles
+shps <- query_sb_datatype("Shapefile")
+head(shps, 2)
+
+# find raster data
+sbraster <- query_sb_datatype("Raster")
+head(sbraster, 2)
+```
+
+## Best of both methods
+
+Although you can query from R, sometimes it's useful to look an item on the web interface. You can use the `query_sb_*` functions and then follow that URL to view items on the web. This is especially handy for viewing maps and metadata, or to check or repair a ScienceBase item if any of the `sbtools`-based commands have failed.
+
+```{r}
+sbmaps <- query_sb_datatype("Static Map Image", limit=3)
+oneitem <- sbmaps[[1]]
+
+# get and open URL from single sbitem
+url_oneitem <- oneitem[['link']][['url']]
+browseURL(url_oneitem)
+
+# get and open URLs from many sbitems
+lapply(sbmaps, function(sbitem) {
+  url <- sbitem[['link']][['url']]
+  browseURL(url)
+  return(url)
+})
+```
+
+## No results
+
+Some of your queries will probably return no results. When there are no results that match your query, the returned list will have a length of 0.
+
+```{r}
+# search for items related to a Water Quality Portal paper DOI
+query_results <- query_sb_doi(doi = '10.1002/2016WR019993')
```

Comment:
lol. making good use of a dead end from your work on the `query_sb_doi` section, eh? there's no chance that this paper will ever make its way onto SB, is there?